### PR TITLE
Apply "silent code cleanup" from IntelliJ

### PR DESCRIPTION
**Pull request by Baltoli**
This PR applies all automatic cleanups that IntelliJ is confident enough to apply silently with no further user input; a rough summary of the different kinds of change are:
* Removal of redundant casts (`T t = (T) v` where `v` already has type `T`).
* Removal of unnecessary calls to `toString()`
* Insertion of `final` where possible
* Collapsing of redundant boolean conditions `if (cond) return true else return false`
* Syntactic fixes (`;;` etc.)
* Use of `StandardCharsets.UTF_8` to simplify exception checks when encoding strings
* Removal of redundant explicit key checks when removing values from hashmaps

As ever, I'm happy to bikeshed these changes - I can redo the analysis and disable some subset of the checks if anyone has objections to the specific results we get.

~~To avoid merge conflicts, this PR is based on #3746 and should only be merged when that PR is merged.~~

### CODE REVIEW

File: kernel/src/main/java/org/kframework/compile/GenerateSentencesFromConfigDecl.java [hunk 1]
```diff
@@ -398,8 +398,8 @@ private static Tuple4<Set<Sentence>, Sort, K, Boolean> computeSentencesOfWellFor
                     fragmentItems.add(NonTerminal(childOptSort));
 
                     sentences.add(Production(Seq(), childOptSort, List(NonTerminal(childSort))));
-                    if (!m.definedKLabels().contains(KLabel("no"+childSort.toString()))) {
-                        sentences.add(Production(KLabel("no"+childSort.toString()), childOptSort, List(Terminal("no"+childSort.toString())),
+                    if (!m.definedKLabels().contains(KLabel("no"+ childSort))) {
+                        sentences.add(Production(KLabel("no"+ childSort), childOptSort, List(Terminal("no"+ childSort)),
```
Comment (radumereuta) [inline kernel/src/main/java/org/kframework/compile/GenerateSentencesFromConfigDecl.java]: Spaces before and after `+`?

File: kernel/src/main/java/org/kframework/compile/SortCells.java [hunk 2]
```diff
@@ -784,7 +784,7 @@ public void apply(KApply k) {
                 }
                 @Override
                 public void apply(KVariable var) {
-                    if (!inKCell && cellFragmentVars.containsKey(var)) {
+                    if (!inKCell) {
```
Comment (radumereuta) [inline kernel/src/main/java/org/kframework/compile/SortCells.java:787]: What?! Why was this removed?

**Review by radumereuta — APPROVED** [review]
Overall, it looks good. I added a question though. Please double check.

File: kernel/src/main/java/org/kframework/compile/SortCells.java (hunk 2 above)
Comment (Baltoli) [reply kernel/src/main/java/org/kframework/compile/SortCells.java:787]: From the [Java documentation](https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html#remove-java.lang.Object-), you don't need to check if a key is present before removing it from a `HashMap`, so the explicit check is redundant.

File: kernel/src/main/java/org/kframework/compile/GenerateSentencesFromConfigDecl.java (hunk 1 above)
Comment (Baltoli) [reply kernel/src/main/java/org/kframework/compile/GenerateSentencesFromConfigDecl.java]: Yep, I'll fix this

Comment (Baltoli) [inline kernel/src/main/java/org/kframework/compile/GenerateSentencesFromConfigDecl.java]:
```suggestion
                    if (!m.definedKLabels().contains(KLabel("no" + childSort))) {
                        sentences.add(Production(KLabel("no" + childSort), childOptSort, List(Terminal("no" + childSort)),
```
